src/main/java/dev/blaauwendraad/masker/json/config/KeyMaskingConfig.java

```diff
@@ -158,130 +74,131 @@ private Builder() {
          * <p>
          * Masking strings with {@literal "***"} is the default behaviour if no string masking option is set.
          *
-         * @see #maskStringCharactersWith(String)
-         *
          * @return the builder instance
+         * @see #maskStringCharactersWith(String)
+         * @see #maskStringsWith(ValueMasker)
          */
         public Builder maskStringsWith(String value) {
-            if (maskStringsWith != null) {
-                throw new IllegalArgumentException("'maskStringsWith(String)' was already set");
-            }
-            checkMutuallyExclusiveStringMaskingOptions();
-            maskStringsWith = value;
+            maskStringsWith(ValueMasker.maskWith(value));
             return this;
         }
 
         /**
          * Mask all characters of string values with the provided character, preserving the length.
          * For example, {@literal "maskMe": "secret" -> "maskMe": "******"}.
          *
+         * @return the builder instance
          * @see #maskStringsWith(String)
+         * @see #maskStringsWith(ValueMasker)
+         */
+        public Builder maskStringCharactersWith(String value) {
+            maskStringsWith(ValueMasker.maskStringCharactersWith(value));
+            return this;
+        }
+
+        /**
+         * Mask all string values with the provided {@link ValueMasker}.
          *
          * @return the builder instance
+         * @see #maskStringsWith(String)
+         * @see #maskStringCharactersWith(String)
          */
-        public Builder maskStringCharactersWith(String value) {
-            if (maskStringCharactersWith != null) {
-                throw new IllegalArgumentException("'maskStringCharactersWith(String)' was already set");
+        public Builder maskStringsWith(ValueMasker valueMasker) {
+            if (maskStringsWith != null) {
+                throw new IllegalArgumentException("'maskStringsWith' was already set");
             }
-            checkMutuallyExclusiveStringMaskingOptions();
-            maskStringCharactersWith = value;
+            maskStringsWith = valueMasker;
             return this;
         }
 
         /**
          * Disables number masking.
          *
+         * @return the builder instance
          * @see #maskNumbersWith(String)
          * @see #maskNumbersWith(int)
          * @see #maskNumberDigitsWith(int)
-         *
-         * @return the builder instance
+         * @see #maskNumbersWith(ValueMasker)
          */
         public Builder disableNumberMasking() {
-            if (disableNumberMasking != null) {
-                throw new IllegalArgumentException("'disableNumberMasking()' was already set");
-            }
-            checkMutuallyExclusiveNumberMaskingOptions();
-            disableNumberMasking = true;
+            maskNumbersWith(ValueMasker.noop());
             return this;
         }
```

Comment:
I'm thinking of getting rid of `disableNumberMasking` / `disableBooleanMasking`, this now can be achieved with `ValueMasker.noop()` and it's not significant enough to have a separate API for that

Reply:
Makes total sense to me.